overlord/configstate/config/helpers.go

```diff
@@ -119,6 +120,90 @@ func GetFromChange(snapName string, subkeys []string, pos int, config map[string
 	return GetFromChange(snapName, subkeys, pos+1, configm, result)
 }
 
+// StoreConfigurationSnapshotMaybe makes a copy of config -> snapSnape configuration into the versioned config.
+// It doesn't do anything if there is no configuration for given snap in the state.
+func StoreConfigurationSnapshotMaybe(st *state.State, snapName string, rev snap.Revision) error {
+	var config map[string]interface{}                     // snap => configuration
+	var configSnapshots map[string]map[string]interface{} // snap => revision => configuration
+
+	// Get current configuration of the snap from state
+	err := st.Get("config", &config)
+	if err == state.ErrNoState {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("internal error: cannot unmarshal configuration: %v", err)
+	}
+	if snapcfg, ok := config[snapName]; ok {
```

Comment:
You could move this to a separate line and reverse the condition to make the code less indented.
```
snapcfg, ok := config[snapName]
if !ok {
  return nil
}
```

Though just a nitpick, it changes nothing.